src/blocks/nvidia_gpu.rs

```diff
@@ -324,115 +342,117 @@ impl Block for NvidiaGpu {
+                                fan_widget.set_state(State::Warning);
+                            } else {
+                                Command::new("nvidia-settings")
+                                    .args(&[
+                                        "-a",
+                                        &format!("[gpu:{}]/GPUFanControlState=0", self.gpu_id),
+                                    ])
+                                    .output()
+                                    .block_error("gpu", "Failed to execute nvidia-settings.")?;
+                                fan_widget.set_state(State::Idle);
+                            }
+                        } else if self.fan_speed_controlled {
                             Command::new("nvidia-settings")
                                 .args(&[
-                                    "-a",
-                                    &format!("[gpu:{}]/GPUFanControlState=1", self.gpu_id),
                                     "-a",
                                     &format!(
                                         "[fan:{}]/GPUTargetFanSpeed={}",
-                                        self.gpu_id, self.fan_speed
+                                        self.gpu_id, new_fan_speed
                                     ),
                                 ])
                                 .output()
                                 .block_error("gpu", "Failed to execute nvidia-settings.")?;
-                            fan_widget.set_text(format!("{:02}%", self.fan_speed));
-                            fan_widget.set_state(State::Warning);
-                        } else {
-                            Command::new("nvidia-settings")
-                                .args(&[
-                                    "-a",
-                                    &format!("[gpu:{}]/GPUFanControlState=0", self.gpu_id),
-                                ])
-                                .output()
-                                .block_error("gpu", "Failed to execute nvidia-settings.")?;
-                            fan_widget.set_state(State::Idle);
+                            self.fan_speed = new_fan_speed;
+                            fan_widget.set_text(format!("{:02}%", new_fan_speed));
                         }
-                    } else if self.fan_speed_controlled {
-                        Command::new("nvidia-settings")
-                            .args(&[
-                                "-a",
-                                &format!(
-                                    "[fan:{}]/GPUTargetFanSpeed={}",
-                                    self.gpu_id, new_fan_speed
-                                ),
-                            ])
-                            .output()
-                            .block_error("gpu", "Failed to execute nvidia-settings.")?;
-                        self.fan_speed = new_fan_speed;
-                        fan_widget.set_text(format!("{:02}%", new_fan_speed));
```

Comment:
This huge diff is caused by enclosing the code inside a if statement to check self.clickable. I did not change anything in that code but I had to increase the indentation of the whole block.